src/core/util.js

```diff
@@ -283,4 +285,221 @@ var parseTimeSeriesForC3 = function(graph_data) {
+    var cells = [];
+    var cell;
+    // populate worksheet with table data
+    var column_labels = ["Model Period", "Run", "Min", "Max", "W.Mean", "Median", "W.Std.Dev", "Units" ];
+    var data_keys = ["model_period", "run", "min", "max", "w_mean", "median", "w_stdev", "units" ];
+    var num_data_rows = Object.keys(data).length + 1;
+    var num_data_cols = data_keys.length
+
+    for(var R = 0; R < num_data_rows; ++R) {
+        for(var C = 0; C < num_data_cols; ++C) {
+            // create header row
+            if(R == 0) cell = {v: column_labels[C]};
+            // create cell object: .v is the actual data
+            else cell = {v: data[R-1][data_keys[C]]};
+            if(cell.v === null) continue;
+            // determine the cell type 
+            if(typeof cell.v === 'number') cell.t = 'n';
+            else cell.t = 's';
+            cells.push(cell);
+        }
+    }
+    return { 'num_rows': num_data_rows, 'num_cols': num_data_cols, 'cells': cells };
+}
+
+/*
+    Helper function for exportTableDataToWorksheet, combines summary rows, data column headers, and data into one worksheet
+*/
+var assembleWorksheet = function (summary_cells, data_cells) {
+    var R = 0;
+    var C = 0;
+    var cell;
+    var cell_ref;
+    var ws = {};
+
+    for(R = 0; R < summary_cells.num_rows; ++R) {
+        for(C = 0; C < summary_cells.num_cols; ++C) {
+            cell_ref = XLSX.utils.encode_cell({c:C,r:R});
+            ws[cell_ref] = summary_cells.cells[(R * summary_cells.num_cols) + C];
+        }
+    }
+    for(var data_row = 0; data_row < data_cells.num_rows; ++data_row) {
+        for(var data_col = 0; data_col < data_cells.num_cols; ++data_col) {
+            cell_ref = XLSX.utils.encode_cell({c:data_col,r:R+data_row});
+            ws[cell_ref] = data_cells.cells[(data_row * data_cells.num_cols) + data_col];
+        }
+    }
+    // set combined worksheet range bounds
+    var range = {
+      s: {c:0, r:0}, 
+      e: {
+          c: (summary_cells.num_cols > data_cells.num_cols ? summary_cells.num_cols : data_cells.num_cols), 
+          r: summary_cells.num_rows + data_cells.num_rows + 1}};
+    ws['!ref'] = XLSX.utils.encode_range(range);
+    return ws;
+}
+
+/*
+    Takes current data displayed in the DataTable and contextual data from user input, 
+    creates an XLSX or CSV file, and serves it to the user for download.
+*/
+var exportTableDataToWorksheet = function(data, format) {
+    // create workbook object containing one or more worksheets
+    var wb = {}
+    wb.Sheets = {};
+    wb.SheetNames = [];
+
+    // convert timestep ID (0-16) to string format
+    var times_of_year = ["January", "February", "March", "April", "May", "June", "July", "August", "September", 
+                      "October", "November", "December", "Winter - DJF", "Spring - MAM", "Summer - JJA", 
+                      "Fall - SON", "Annual"];
+    var time_of_year = times_of_year[this.state.dataTableTimeOfYear];
```

Comment:
"Utility" functions are supposed to be atomic and independent of any context. I don't think that using `this` in this function is a good idea.